mypy/expandtype.py

```diff
@@ -370,6 +373,23 @@ def visit_callable_type(self, t: CallableType) -> Type:
                     ret_type=t.ret_type.accept(self),
                     type_guard=(t.type_guard.accept(self) if t.type_guard is not None else None),
                 )
+            # TODO: it seems this only has to be done *sometimes*. Conceptually this should only
+            #       be done once; we should update that "once" location rather than here.
+            #       (see testAlreadyExpandedCallableWithParamSpecReplacement)
+            elif isinstance(repl, ParamSpecType) and len(t.arg_types) == 2:
+                # We're substituting one paramspec for another; this can mean that the prefix
+                # changes. (e.g. sub Concatenate[int, P] for Q)
+                prefix = repl.prefix
+                old_prefix = param_spec.prefix
+
+                # Check assumptions. I'm not sure what order to switch these:
+                assert not old_prefix.arg_types or not prefix.arg_types
```

Comment:
This looks a bit suspicious to me. Why can't both the arg types truthy?

Reply:
I haven't found a case where they were both truthy and yet not equal to each other yet -- I don't know what order to mix their arguments together. My gut says what I wrote next is correct but I would rather have an explicit test case (or be shown that this never happens)